src/sound/YMF262.cc

```diff
@@ -1045,75 +1045,81 @@ void YMF262::writeRegDirect(unsigned r, uint8_t v, EmuTime::param time)
 {
 	reg[r] = v;
 
-	switch (r) {
-	case 0x104:
-		// 6 channels enable
-		channel[ 0].extended = (v & 0x01) != 0;
-		channel[ 1].extended = (v & 0x02) != 0;
-		channel[ 2].extended = (v & 0x04) != 0;
-		channel[ 9].extended = (v & 0x08) != 0;
-		channel[10].extended = (v & 0x10) != 0;
-		channel[11].extended = (v & 0x20) != 0;
-		return;
-
-	case 0x105:
-		// OPL3 mode when bit0=1 otherwise it is OPL2 mode
-		OPL3_mode = v & 0x01;
-
-		// Verified on real YMF278: When NEW2 bit is first set, a read
-		// from the status register (once) returns bit 1 set (0x02).
-		// This only happens once after reset, so clearing NEW2 and
-		// setting it again doesn't cause another change in the status
-		// register. Also, only bit 1 changes.
-		if ((v & 0x02) && !alreadySignaledNEW2 && isYMF278) {
-			status2 = 0x02;
-			alreadySignaledNEW2 = true;
-		}
-
-		// following behaviour was tested on real YMF262,
-		// switching OPL3/OPL2 modes on the fly:
-		//  - does not change the waveform previously selected
-		//    (unless when ....)
-		//  - does not update CH.A, CH.B, CH.C and CH.D output
-		//    selectors (registers c0-c8) (unless when ....)
-		//  - does not disable channels 9-17 on OPL3->OPL2 switch
-		//  - does not switch 4 operator channels back to 2
-		//    operator channels
-		return;
-	}
-
 	unsigned ch_offset = (r & 0x100) ? 9 : 0;
 	switch (r & 0xE0) {
 	case 0x00: // 00-1F:control
-		switch (r & 0x1F) {
-		case 0x01: // test register
-			break;
-
-		case 0x02: // Timer 1
-			timer1->setValue(v);
-			break;
-
-		case 0x03: // Timer 2
-			timer2->setValue(v);
-			break;
-
-		case 0x04: // IRQ clear / mask and Timer enable
-			if (v & 0x80) {
-				// IRQ flags clear
-				resetStatus(0x60);
-			} else {
-				changeStatusMask((~v) & 0x60);
-				timer1->setStart((v & R04_ST1) != 0, time);
-				timer2->setStart((v & R04_ST2) != 0, time);
+		if ((r & 0x100) == 0) {
+			switch (r & 0x1F) {
```

Comment:
Thanks a lot for your PR.
May I suggest one small tweak:
* Remove the test for `((r & 0x100) == 0)`.
* And instead do a `switch (r)` (instead of `switch (r & 0x1f)`), which then handles both 0x000-0x01f and 0x100-0x11f.

The main advantage is a reduction in indentation level.